backend/src/models/user.ts

```diff
@@ -175,11 +197,12 @@ export class User {
       lastName: "last_name",
       isAdmin: "is_admin",
     });
-    const idVarIdx = "$" + (values.length + 1);
+
+    const usernameVarIdx = "$" + (values.length + 1);
 
     const querySql = `UPDATE users
             SET ${setCols} 
-            WHERE id = ${idVarIdx} 
+            WHERE username = ${usernameVarIdx} 
```

Comment:
If a user is updating their username in the same request, wouldn't this cause an error as the `WHERE` condition won't be able to find an existing user with the updated username?